# Summarizing Automobile Evaluation Data

In the following project you’ll use what you’ve learned about summarizing categorical data to analyze a sample from a popular open source dataset. This dataset contains information on the cost and physical attributes of several thousand cars. Originally, this dataset was used for to train a classification model that assigned an acceptability score/category to cars based on these attributes.

In [60]:
import numpy as np
import pandas as pd
pd.set_option('display.max_colwidth', None)
pd.set_option('display.max_rows', None)

In [61]:
car_eval = pd.read_csv('car-eval-dataset.csv')
car_eval.head()

buying_cost maintenance_cost doors capacity luggage safety acceptability  \
0       vhigh              low     4        4   small    med         unacc   
1       vhigh              med     3        4   small   high           acc   
2         med             high     3        2     med   high         unacc   
3         low              med     4     more     big    low         unacc   
4         low             high     2     more     med   high           acc   

  manufacturer_country  
0                China  
1               France  
2        United States  
3        United States  
4          South Korea

### Summarizing Manufacturing Country

1.
manufacturer_country is a nominal categorical variable that indicates the country of the manufacturer of each car reviewed. Create a table of frequencies of all the cars reviewed by manufacturer_country. What is the modal category? Which country appears 4th most frequently? Print out your results.

In [62]:
car_frequency = car_eval['manufacturer_country'].value_counts()
car_frequency

Japan            228
Germany          218
South Korea      159
United States    138
Italy             97
France            87
China             73
Name: manufacturer_country, dtype: int64

In [63]:
car_frequency.index[3]

'United States'

2.
Calculate a table of proportions for countries that appear in manufacturer_country in the dataset. What percentage of cars were manufactured in Japan?

In [64]:
car_eval['manufacturer_country'].unique()

array(['China', 'France', 'United States', 'South Korea', 'Italy',
       'Japan', 'Germany'], dtype=object)

In [65]:
car_proportion = car_eval['manufacturer_country'].value_counts(normalize=True, dropna=False)
car_proportion

Japan            0.228
Germany          0.218
South Korea      0.159
United States    0.138
Italy            0.097
France           0.087
China            0.073
Name: manufacturer_country, dtype: float64

The cars were manufactures in Japan is 22.8%

### Summarizing Buying Costs

3.
buying_cost is a categorical variable which describes the cost of buying any car in the dataset. Print out a list of the possible values for this variable.

In [83]:
car_eval['buying_cost'].unique()

['vhigh', 'med', 'low', 'high']
Categories (4, object): ['low' < 'med' < 'high' < 'vhigh']

4.
buying_cost is an ordinal categorical variable, which means we can create an order associated with the values in the data and perform additional numeric operations on the variable. Create a new list, buying_cost_categories, that contains the unique values in buying_cost, ordered from lowest to highest.

In [84]:
buying_cost_categories = ['low', 'med', 'high', 'vhigh']

5.
Convert buying_cost to type 'category' using the list you created in the previous exercise.

In [85]:
car_eval['buying_cost'] = pd.Categorical(car_eval['buying_cost'], buying_cost_categories, ordered=True)
car_eval['buying_cost'].unique()

['vhigh', 'med', 'low', 'high']
Categories (4, object): ['low' < 'med' < 'high' < 'vhigh']

.
Calculate the median category of the buying_cost variable and print the result.

In [86]:
median_index = np.median(car_eval['buying_cost'].cat.codes)
median_index

1.0

In [87]:
median_car_status = buying_cost_categories[int(median_index)]
median_car_status

'med'

### Summarizing Luggage Capacity

7.
luggage is a categorical variable in the car evaluations dataset that records the luggage capacity for each reviewed car. Calculate a table of proportions for this variable and print the result.

In [88]:
car_eval['luggage'].value_counts(normalize=True)

small    0.339
med      0.333
big      0.328
Name: luggage, dtype: float64

8.
Are there any missing values in this column? Replicate the table of proportions from the previous exercise, but do not drop any missing values from the count. Print your result.

In [89]:
car_eval['luggage'].value_counts(dropna=False, normalize=True)

small    0.339
med      0.333
big      0.328
Name: luggage, dtype: float64

9.
Without passing normalize = True to .value_counts(), can you replicate the result you got in the previous exercises

In [90]:
car_eval['luggage'].value_counts()/len(car_eval['luggage'])

small    0.339
med      0.333
big      0.328
Name: luggage, dtype: float64

### Summarizing Passenger Capacity 

10.
doors is a categorical variable in the car evaluations dataset that records the count of doors for each reviewed car. Find the count of cars that have 5 or more doors. You can identify cars with 5+ doors by looking for cars that have a value of '5more' in the doors column. Print your result.

In [93]:
car_eval['doors'].unique()

array(['4', '3', '2', '5more'], dtype=object)

In [91]:
more_five_doors = np.sum(car_eval['doors'] >= '5more')
more_five_doors

246

11.
Find the proportion of cars that have 5+ doors and print the result.

In [94]:
five_car_proportion = (car_eval['doors'] == '5more').mean()
five_car_proportion

0.246

In [95]:
car_eval['doors'].value_counts(normalize=True, dropna=False)

4        0.263
3        0.252
5more    0.246
2        0.239
Name: doors, dtype: float64